### PR TITLE
Implement custom serialization for `Opening`

### PR DESCRIPTION
**Pull request by moCello**
Resolves: #55

### CODE REVIEW

File: src/opening.rs [hunk 1]
```diff
@@ -91,6 +94,80 @@ where
...
+
+    /// Serialize an [`opening`] to a vector of bytes.
+    // Once the new implementation of the `Serializable` trait becomes
+    // available, we will want that instead, but for the time being we use
+    // this implementation.
+    pub fn to_var_bytes<const T_SIZE: usize>(&self) -> Vec<u8>
+    where
+        T: Serializable<T_SIZE>,
+    {
+        let mut bytes = Vec::new();
```
Comment (ureeves) [inline src/opening.rs:106]: Could we at least pre-allocate the number of bytes required to minimize the number of allocations? It seems to me it is going to be:

```text
root:      T_SIZE 
branch:    T_SIZE * H * A
positions: u32::SIZE * H

opening:   T_SIZE + (T_SIZE * A + u32::SIZE) * H
```

```suggestion
        let mut bytes = Vec::with_capacity(T_SIZE + (T_SIZE * A + u32::SIZE) * H);
```

File: src/opening.rs [hunk 2]
```diff
@@ -91,6 +94,80 @@ where
...
+    // this implementation.
+    pub fn from_slice<const T_SIZE: usize>(
+        buf: &[u8],
+    ) -> Result<Self, BytesError>
+    where
+        T: Serializable<T_SIZE>,
+        <T as Serializable<T_SIZE>>::Error: dusk_bytes::BadLength,
+        dusk_bytes::Error: From<<T as Serializable<T_SIZE>>::Error>,
+    {
+        let mut bytes = buf;
```
Comment (ureeves) [inline src/opening.rs:145]: Since we know the size here we could also return a `BytesError::BadLength` if the size is not what we expect.

**Review by ureeves — CHANGES_REQUESTED** [review]
Some changes since we actually know the size of an opening

File: src/opening.rs (hunk 2 above)
Comment (moCello) [reply src/opening.rs]: This is a really good suggestion as it actually makes the test fail :sweat_smile: 
Will investigate why the lengths don't match

Comment (moCello) [reply src/opening.rs]: My bad, the positions are encoded as `u32` (`Serializable` is not implemented for `usize`), so the expected length was calculated incorrectly. 
Everything should work now!

**Review by ureeves — APPROVED** [review]
LGTM

**Pull request merged**